lib/queue.js

```diff
@@ -92,55 +92,64 @@ var Queue = function Queue(name, url, opts){
     throw Error('Options must be a valid object');
   }
 
-  var redisOpts = opts.redis || {};
+  this.name = name;
+  this.token = uuid();
 
-  _.defaults(redisOpts, {
+  opts.redis = opts.redis || {};
+
```

Comment:
as mentioned in the previous PR, we need to clone opts and opts.redis, or we will be modifying the user's options object which could create side effects.